librz/io/p/io_SREC.c

```diff
@@ -0,0 +1,810 @@
+					}
+					cksum += byte;
+				}
+				sec_size += bc;
+				next_addr += bc;
+				if (eol) {
+					// checksum
+					if (sscanf(str + 12 + (i * 2), "%02x", &byte) != 1) {
+						eprintf("unparsable data !\n");
+						goto fail;
+					}
+					cksum = ~(cksum & 0xff); //this might be useless cause cksum is ut8
+					if (cksum != byte) {
+						eprintf("Checksum failed (got %02x expected %02x)\n", byte, cksum);
+						goto fail;
+					}
+					*eol = 'S';
+				}
+				str = eol;
+				break;
+
+			case 7: //same with S9 but on 32 bits
+				
+				// S / 7 / (addres+data+checksum) bytes / address / - / checksum
+				l = sscanf(str + 4, "%08x", &addr_tmp);			
+				addr_tmp &= 0xffffffff;
+				
+				eol = strchr(str + 1, 'S');
+				if (eol) {
+					*eol = 0;
+				}
+
+				cksum = bc;
+				cksum += addr_tmp >> 8;
+				cksum += addr_tmp >> 16;
+				cksum += addr_tmp >> 32;
+				cksum += addr_tmp & 0xff;
+
+				if ((next_addr != addr_tmp) || ((sec_size + bc) > SEC_MAX)) {
+					//previous block is not contiguous, or
+					//section buffer is full => write a sparse chunk
+					if (sec_size && sec_size < UT16_MAX) {
+						if (rz_buf_write_at(rbuf, sec_start, sec_tmp, (int)sec_size) != sec_size) {
+							eprintf("sparse buffer problem, giving up\n");
+							goto fail;
+						}
+					}
+					//advance cursor, reset section
+					sec_start = segreg + addr_tmp;
+					next_addr = addr_tmp;
+					sec_size = 0;
+				}
+				sec_size += bc;
+				next_addr += bc;
+				if (eol) {
+					// checksum
+					if (sscanf(str + 12 + (i * 2), "%02x", &byte) != 1) {
+						eprintf("unparsable data !\n");
+						goto fail;
+					}
+					cksum = ~(cksum & 0xff); //this might be useless cause cksum is ut8
+					if (cksum != byte) {
+						eprintf("Checksum failed (got %02x expected %02x)\n", byte, cksum);
+						goto fail;
+					}
+					*eol = 'S';
+				}
+				str = eol;
+				break;
+
+			case 4: //reserved
+			}
+	} while (str);
+	free(sec_tmp);
+	return true;
+fail:
+	free(sec_tmp);
+	return false;
+}
+
+static RzIODesc *__open(RzIO *io, const char *pathname, int rw, int mode) {
+	Rihex *mal = NULL;
+	char *str = NULL;
+	if (__plugin_open(io, pathname, 0)) {
+		str = rz_file_slurp(pathname + 7, NULL);
+		if (!str) {
+			return NULL;
+		}
+		mal = RZ_NEW0(Rihex);
+		if (!mal) {
+			free(str);
+			return NULL;
+		}
+		mal->rbuf = rz_buf_new_sparse(io->Oxff);
+		if (!mal->rbuf) {
+			free(str);
+			free(mal);
+			return NULL;
+		}
+		if (!SREC_parse(mal->rbuf, str)) {
+			eprintf("ihex: failed to parse file\n");
+			free(str);
+			rz_buf_free(mal->rbuf);
+			free(mal);
+			return NULL;
+		}
+		free(str);
+		return rz_io_desc_new(io, &rz_io_plugin_ihex,
+			pathname, rw, mode, mal);
+	}
+	return NULL;
+}
+
+static bool __resize(RzIO *io, RzIODesc *fd, ut64 size) {
+	if (!fd) {
+		return false;
+	}
+	Rihex *rih = fd->data;
+	if (rih) {
+		return rz_buf_resize(rih->rbuf, size);
+	}
+	return false;
+}
+
+RzIOPlugin rz_io_plugin_ihex = {
+	.name = "ihex",
```

Comment:
Why `ihex`? Should be `srec` obviously. And everywhere else in this file. Also better to call it `io_srec.c` instead.

Reply:
You forgot to remove that `io_SREC.c` file from the PR.